src/library/ButtonGroup/__tests__/ButtonGroup.spec.js

```diff
@@ -0,0 +1,104 @@
+/* @flow */
+import React from 'react';
+import { shallow } from 'enzyme';
+import { mountInThemeProvider } from '../../../../utils/enzymeUtils';
+import Button from '../../../library/Button';
+import ButtonGroup from '../ButtonGroup';
+import examples from '../../../website/app/demos/ButtonGroup/examples';
+import testDemoExamples from '../../../../utils/testDemoExamples';
+
+const defaultProps = {
+  'aria-label': 'Test',
+  children: <Button>Content</Button>,
+  onChange: jest.fn(),
+  onClick: jest.fn()
+};
+
+function shallowButtonGroup(props = {}) {
+  const buttonGroupProps = {
+    ...defaultProps,
+    ...props
+  };
+  return shallow(<ButtonGroup {...buttonGroupProps} />);
+}
+
+function mountButtonGroup(props = {}) {
+  const buttonGroupProps = {
+    ...defaultProps,
+    ...props
+  };
+
+  return mountInThemeProvider(<ButtonGroup {...buttonGroupProps} />);
+}
+
+describe('ButtonGroup', () => {
+  testDemoExamples(examples);
+
+  it('renders', () => {
+    const buttonGroup = shallowButtonGroup();
+
+    expect(buttonGroup.exists()).toEqual(true);
+  });
+
+  describe('event handlers', () => {
+    let button, buttonGroup;
+
+    beforeEach(() => {
+      defaultProps.onClick.mockReset();
+      defaultProps.onChange.mockReset();
+    });
+
+    describe('when no mode', () => {
+      beforeEach(() => {
+        [, buttonGroup] = mountButtonGroup();
+        button = buttonGroup.find('button');
+      });
+
+      it('calls onClick handler', () => {
+        button.simulate('click');
+
+        expect(defaultProps.onClick).toHaveBeenCalled();
+      });
+    });
+
+    describe('when mode=checkbox', () => {
+      beforeEach(() => {
+        [, buttonGroup] = mountButtonGroup({ mode: 'checkbox' });
+        button = buttonGroup.find('button');
+      });
+
+      it('calls onClick handler', () => {
+        button.simulate('click');
+
+        expect(defaultProps.onClick).toHaveBeenCalled();
+      });
+
+      it('calls onChange handler', () => {
+        button.simulate('click');
+        button.simulate('click');
+
+        expect(defaultProps.onChange).toHaveBeenCalledTimes(2);
+      });
+    });
+
+    describe('when mode=radio', () => {
+      beforeEach(() => {
+        [, buttonGroup] = mountButtonGroup({ mode: 'radio' });
+        button = buttonGroup.find('button');
+      });
+
+      it('calls onClick handler', () => {
+        button.simulate('click');
+
+        expect(defaultProps.onClick).toHaveBeenCalled();
+      });
+
+      it('calls onChange when Button is toggled', () => {
+        button.simulate('click');
+        button.simulate('click');
+
+        expect(defaultProps.onChange).toHaveBeenCalledTimes(1);
```

Comment:
I think some of these test cases can&should be compressed. many of the `it`s are doing similar things with different `expects`.

`button.simulate('click')` looks like set up to me and should probably be up in the beforeEach. then the assertions can be in the different it blocks.

what else should we test besides the event handlers?